src/client/cli/cmd/info.cpp

```diff
@@ -29,14 +29,16 @@ namespace
 // TODO@snapshots move this to common_cli once required by other commands
 std::vector<mp::InstanceSnapshotPair> add_instance_and_snapshot_names(const mp::ArgParser* parser)
 {
-    std::vector<mp::InstanceSnapshotPair> instance_snapshot_names(parser->positionalArguments().count());
+    std::vector<mp::InstanceSnapshotPair> instance_snapshot_names;
+    instance_snapshot_names.reserve(parser->positionalArguments().count());
 
     for (const auto& arg : parser->positionalArguments())
     {
         mp::InstanceSnapshotPair inst_snap_name;
-        inst_snap_name.set_instance_name(arg.left(arg.indexOf('.')).toStdString());
-        if (arg.indexOf('.') >= 0)
-            inst_snap_name.set_snapshot_name(arg.right(arg.length() - arg.indexOf('.') - 1).toStdString());
+        auto index = arg.indexOf('.');
+        inst_snap_name.set_instance_name(arg.left(index).toStdString());
+        if (index >= 0)
+            inst_snap_name.set_snapshot_name(arg.right(arg.length() - index - 1).toStdString());
```

Comment:
Sorry, I still think this should verify the format before sending it over to the daemon. The client is parsing it anyway, so it already needs to understand the structure. And otherwise errors can be confusing:

```
$ mp info ..
info failed: The following errors occurred:
instance "" does not exist
```

Reply:
Actually, let's leave this for a follow up PR. The argument should only be parsed when introducing snapshot details anyway. And then there are other things that need to be changed, like the help and other error messages. For instance:

```
$ mp info
Name argument or --all is required
```